### PR TITLE
Make generic task dropdown

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects #7102 

### Description
First PR to fix the task dropdown. This change adds a new task drop down that will be used for all cases. We replace the attorney dropdown with this new one, and set up the structures to add new options.

We also move checkout flows to have unique URL paths to move state from Redux into the URL about which flow we're in.

### Testing Plan
1. Manually test various checkout flows.

### CODE REVIEW

File: app/models/legacy_tasks/legacy_task.rb [hunk 1]
```diff
@@ -47,6 +47,18 @@ def days_waiting
     (Time.zone.today - assigned_at.to_date).to_i if assigned_at
   end
 
+  def get_allowed_actions(role)
+    puts "TEST", role, Constants::TaskActionList::LEGACY_ACCESS_CONTROL.first[:vacols_role].include?(role)
+    Constants::TaskActionList::LEGACY_ACCESS_CONTROL.
```
Comment (mdbenjam) [inline app/models/legacy_tasks/legacy_task.rb]: Legacy tasks will continue to be determined by role 🤷‍♂️

File: client/app/queue/CaseSnapshot.jsx [hunk 2]
```diff
@@ -254,17 +255,18 @@ export class CaseSnapshot extends React.PureComponent<Props> {
...
       userRole
     } = this.props;
-    let CheckoutDropdown = <React.Fragment />;
+    let ActionDropdown;
     const dropdownArgs = { appealId: appeal.externalId };
 
+
     if (userRole === USER_ROLE_TYPES.attorney) {
-      CheckoutDropdown = <AttorneyActionsDropdown {...dropdownArgs} />;
+      ActionDropdown  = <ActionsDropdown task={taskAssignedToUser} appealId={appeal.externalId} />;
```
Comment (mdbenjam) [inline client/app/queue/CaseSnapshot.jsx]: In this PR we're only replacing the attorney dropdown. The others to follow in subsequent PRs!

File: client/app/queue/utils.js [hunk 3]
```diff
@@ -332,7 +330,7 @@ export const buildCaseReviewPayload = (
   }
 
   if (userRole === USER_ROLE_TYPES.attorney) {
-    _.extend(payload.data.tasks, { document_type: decision.type });
+    _.extend(payload.data.tasks, { document_type: checkoutFlow });
```
Comment (mdbenjam) [inline client/app/queue/utils.js]: We no longer get the type from decision.type but instead encode it in the URL.

File: lib/constants/task_action_list.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+module Constants::TaskActionList
+  ACCESS_CONTROL = [
+    {
+      task_type: ["AttorneyTask"],
```
Comment (mdbenjam) [inline lib/constants/task_action_list.rb]: We can add more conditions here in the future. I'm thinking we'll need to add things css_role, task_action, and possibly others.

File: client/app/constants/SpecialIssues.js [hunk 5]
```diff
@@ -72,7 +72,8 @@ const SPECIAL_ISSUES = [
     specialIssue: 'hearingIncludingTravelBoardVideoConference',
     stationOfJurisdiction: 'regional',
     snakeCase: 'hearing_including_travel_board_video_conference',
-    unhandled: null
+    unhandled: null,
+    nonCompensation: true
```
Comment (mdbenjam) [inline client/app/constants/SpecialIssues.js]: Unrelated change. Just wanted to get it in: https://github.com/department-of-veterans-affairs/caseflow/issues/6903#issuecomment-420700931

File: lib/constants/task_action_list.rb (hunk 4 above)
Comment (lowellrex) [reply lib/constants/task_action_list.rb]: Would it be possible to move this access control to the various Task classes themselves? Perhaps calling something like `generic_task.available_actions(current_user)` would return this list. One advantage I see in making these instance methods is that it allows us to be flexible around which actions are returned based on the current state of the task itself.
